sw/host/opentitanlib/src/image/image.rs

```diff
@@ -98,6 +99,24 @@ impl Image {
         Ok(())
     }
 
+    /// Updates the signature field in the `Manifest`.
+    pub fn update_signature(&mut self, signature: Signature) -> Result<()> {
+        let manifest = self.borrow_manifest_mut()?;
+        let mut manifest_def: ManifestDef = (&*manifest).try_into()?;
+        manifest_def.update_signature(ManifestRsa::from_le_bytes(signature.to_le_bytes())?);
+        *manifest = manifest_def.try_into()?;
```

Comment:
Can you comment why we need this conversion from `Manifest` to `ManifestDef` and back? Also applies to `update_modulus()` below.